scripts/FigS6_celldex-ref-comparison.R

```diff
@@ -0,0 +1,192 @@
+  singler_results <- SingleR::classifySingleR(
+    trained = singler_model,
+    test = sce,
+    fine.tune = TRUE
+  )
+  return(singler_results)
+}
+
+create_delta_median_df <- function(singler_results){
+  
+  # extract scores into matrix
+  singler_scores <- singler_results$scores
+  
+  # Create data frame for plotting with delta median and the full *non-pruned* cell labels
+  delta_median_df <- tibble::tibble(
+    delta_median = rowMaxs(singler_scores) - rowMedians(singler_scores),
+    # Need to grab the non-pruned label for this plot
+    full_labels = singler_results$labels,
+    # if pruned.labels are NA ==> low confidence
+    # so, negate for this variable:
+    confident = !is.na(singler_results$pruned.labels)
+  ) |>
+    dplyr::mutate(
+      confident = ifelse(confident, "High-quality", "Low-quality")
+    )
+  
+  return(delta_median_df)
+}
+
+all_delta_df <- all_sce_objects |> 
+  # apply to all objects 
+  purrr::map(\(sce) {
+    # for each singleR model, run SingleR and grab delta median 
+    delta_median_df <- singler_model_list |> 
+      purrr::map(\(singler_model){
+        
+        # get singleR results 
+        delta_median_df <- run_singler(
+          sce = sce,
+          singler_model = singler_model
+        ) |> 
+          # get delta median and confident labels
+          create_delta_median_df()
+      }) |> 
+      # combine by reference 
+      dplyr::bind_rows(.id = "celldex_reference")
+  }) |>
+  # combine by library id 
+  dplyr::bind_rows(.id = "library_id")
+
+# Plot -------------------------------------------------------------------------
+
+# define diagnosis groups 
+brain_libraries <- c("SCPCL000001", "SCPCL000002", "SCPCL000004")
+blood_libraries <- c("SCPCL000296", "SCPCL000297", "SCPCL000298")
+sarcoma_libraries <- c("SCPCL000478", "SCPCL000484", "SCPCL000488")
+
+# add diagnosis group
+all_delta_df <- all_delta_df |> 
+  dplyr::mutate(
+    diagnosis_group = dplyr::case_when(
+      library_id %in% brain_libraries ~ "Brain and CNS",
+      library_id %in% blood_libraries ~ "Leukemia",
+      library_id %in% sarcoma_libraries ~ "Sarcoma"
+    ),
+    plot_title = glue::glue("{library_id}\n{diagnosis_group}")
+  ) 
+
+# Subset the data to just confident points for median+/-IQR
+delta_median_confident_df <- all_delta_df |>
+  dplyr::filter(confident == "High-quality")
+
+# diagnosis group palette 
+diagnosis_group_palette <- readr::read_tsv(diagnosis_group_palette_file)
+
+# get list of all colors 
+diagnosis_colors <- diagnosis_group_palette$color |> 
+  purrr::set_names(diagnosis_group_palette$diagnosis_group)
+
+# define suspension backgrounds to use on facet strips 
+backgrounds <- rep(c("Brain and CNS", "Leukemia", "Sarcoma"), each = 3) |>
+  purrr::map(\(x) element_rect(fill = diagnosis_colors[[x]]))
+
+# create delta median plot 
+all_delta_plot <- ggplot(all_delta_df, aes(x = celldex_reference, y = delta_median, shape = confident, alpha = confident)) +
+  ggforce::geom_sina(
```

Comment:
The fact that the confident and non-confident distributions have different widths is a bit odd (makes sense, but looks funny). Maybe try `scale = "width"` and see how that looks?

I also wondered if we needed to worry about sorting to make the non-confident cells be on top. I'm not sure exactly whether it would make a difference, but adding a sort on `confident` to `all_delta_df` might be worth a look

Reply:
Just want to confirm that we like using `scale = "width"` as an option? I played around with sorting, but personally I think it all looked worse. 

<img width="530" alt="Screenshot 2024-03-11 at 11 12 23 AM" src="https://github.com/AlexsLemonade/scpca-paper-figures/assets/54039191/fa4f78bd-3836-4497-8e41-33d4105c1328">

Reply:
I might make the width a bit narrower by setting `maxwidth=0.8` or something like that. 

But I realized that the main reason that it doesn't look as I expected is because `geom_sina` does not understand `shape` or `fill` as aesthetics, which means that you can't get them to use the same distribution unless the only difference is color. I like the open circles, so I am torn!

Reply:
I also like the open circles so I think we should keep that feature. Here's with `maxwidth=0.5`. I think I like this option the best, what do we think? 

<img width="617" alt="Screenshot 2024-03-11 at 12 32 24 PM" src="https://github.com/AlexsLemonade/scpca-paper-figures/assets/54039191/30366115-ab17-4e51-9457-f276c93668d8">